drivers/console/aarch64/console.S

```diff
@@ -95,10 +95,15 @@ endfunc console_getc
 	 * data that hasn't been output. It returns 0
 	 * upon successful completion, otherwise it
 	 * returns -1.
-	 * Clobber list : x0, x1
+	 * Clobber list : x0, x1, x2, x3, x4
 	 * ---------------------------------------------
 	 */
 func console_flush
+#if COREBOOT
+	mov	x4, x30
+	bl	coreboot_cbmc_flush
+	mov	x30, x4
+#endif
```

Comment:
> (or if the whole console API was made more flexible and would work by registering function pointers at runtime or something like that)

Yes, that was what I was envisaging. The "splitting" console API would be a struct of function pointers corresponding to the existing console API. This would be much like the PSCI pm-ops API. Each console driver would register a struct containing function pointers unique to that driver. The platform would also provide a macro to determine the number of supported consoles (i.e. how much memory to allocate for the array of structs).

I'm sorry but I disagree that choosing console types is a generic user config option. Each console type implies a particular kind of hardware availability, which the platform should be responsible for defining. The set of potential console types does not look bounded to me and would make the generic code more cluttered and less maintainable.